src/calibre/ebooks/metadata/book/render.py

```
@@ -66,6 +67,19 @@ def search_action_with_data(search_term, value, book_id, field=None, **k):
     return search_action(search_term, value, field=field, book_id=book_id, **k)
 
 
+def cc_search_action_with_data(search_term, value, book_id, fm, mi, field=None,  **k):
+    if mi is not None and fm is not None:
+        template = fm.get('display', {}).get('web_search_template')
+        if template:
+            formatter = SafeFormat()
+            mi.set('item_value', value)
+            u = formatter.safe_format(template, mi, "BOOK DETAILS WEB LINK", mi)
```

Maybe add a try/except to ignore exceptions during template evaluation.